python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -287,11 +322,11 @@ def __init__(
 
             group_name = normalize_group_name(spec.group_name)
 
-            if node_def is not None:
-                output_def, _ = node_def.resolve_output_to_origin(
-                    output_names_by_key[spec.key], None
+            if self._computation is not None:
+                output_def, _ = self._computation.node_def.resolve_output_to_origin(
+                    self._computation.output_names_by_key[spec.key], None
```

Comment:
kwarg would be nice for bare `None` argument

Reply:
agree, though feels orthogonal to this PR. will try to get this done in a followup.